### ISSUE TITLE
A11y_.NETCore3.0(WPF)_WPF ControlsAndLayout_Layout Dock panel_ScreenReader: Screen reader focus is not moving in proper logical order. 

### ISSUE THREAD

**Issue by grubioe** [dotnet/wpf#1470]
Copied from VSO [956760](https://devdiv.visualstudio.com/DevDiv/_queries/edit/956760/)

Check out Accessibility Insights! - Identify accessibility bugs before check-in and make bug fixing faster and easier.”

Environment Details:

Application Name: .NET Core 3.0 (WPF)

.NET Core SDK version:  3.0.100-preview8-013442
Windows Version: Windows10

ScreenReader Version: Narrator, NVDA 2019.1.1

Pre-requisite:

1) Download the latest .NET Core 3 daily build by going to this link.
2) Install Windows (x86 or x64) versions of sdk listed under Master (3.0.x Runtime) by clicking "Installer" link.

Repro Steps:

1) Download the latest .NET Core 3 daily build given on .NET Core daily builds page. 
2) Install Windows (x86 or x64) versions of sdk listed under Master (3.0.x Runtime) by clicking "Installer" link.
3) Run exe file for "ControlsAndLayout" from the build app folder provided. 
4) Main Window for "ControlsAndLayout" solution will appear. 
5) From the Sample Library pane under Layout, select "DockPanel". 
6) Turn on screen reader
7) Navigate to main content.

Actual:

Screen reader focus is not moving in proper logical order. i.e. from left to right and then Top to bottom.

Expected:

Screen reader focus should move in proper logical order/

User Impact:

The screen reader user will get confused  for content ordering.

Recommendations:

Please refer to below URL which is repository for accessibility bug fixes.

https://microsoft.sharepoint.com/teams/msenable/Pages/Home.aspx

MAS Reference:

MAS 1.3.2:

https://microsoft.sharepoint.com/:w:/r/teams/msenable/_layouts/15/WopiFrame.aspx?sourcedoc={1ee227a7-e3c5-47b8-ab18-48ca8595077b}

**Comment by arpitmathur** [comment]
Moving this to 3.1 as this issue does not prevent a user from doing work. Prioritizing this as rank40.

**Comment by arpitmathur** [comment]
The order in which we define the elements in XAML affects how they're laid out in the automation tree.

This particular DockPanel was laid out in the following XAML order:

<DockPanel>
            <Border DockPanel.Dock="Top"/>
            <Border DockPanel.Dock="Top"/>
            <Border DockPanel.Dock="Left"/>
            <Border DockPanel.Dock="Bottom"/>
            <Border/>
 </DockPanel>

This means that when using Narrator, we tab through these elements in exactly the order in which they're defined in the XAML. 

Top -> Top -> Left -> Bottom -> LastChild

This doesn't follow top to bottom, left to right tabbing order.

Also, just swapping the order in which they're laid out in the XAML isn't a possible fix because that would affect how the DockPanel snaps each element, resulting in a different arrangement of the elemens in the DockPanel.

**Comment by weltkante** [comment]
> This doesn't follow top to bottom, left to right tabbing order.

I don't have the example in question so maybe I'm missing an important detail here, but default tab order in WPF is not from top to bottom, left to right either. Thats why there is a `TabIndex` property, to have a tabbing order different from XAML order. Or do you mean to say narrator doesn't respect the TabIndex property if defined? I'd expect narrator to tab in the same order as the user pressing tab keys, and a naive DockPanel (without TabIndex on its children) will tab in order of the children definition, not in "visual layout order".
